balearesg/pragmate-ui
# Refactor and Update Component Documentation and Code Styles

## Description

This PR addresses multiple components within our project to enhance documentation clarity and bring our codebase up to current coding standards. Special attention has been given to the Alert, Breadcrumb, ButtonGroup, Code, Empty, Opinions, Pricing, Progress Bar, and Spinner components. Moreover, we've standardized the interface for several props types and made semantic adjustments, such as renaming `currentValue` to `value` in the Progress Bar component.

## Changes

- **Alert Component**: Reviewed and updated documentation and code style.
- **Breadcrumb Component**: Reviewed and updated documentation and code style.
- **ButtonGroup Component**: Reviewed and updated documentation and code style.
- **Code Component**: Reviewed and updated documentation and code style.
- **Empty Component**: Reviewed and updated documentation and code style.
- **Opinions Component**: Reviewed and updated documentation and code style.
- **Progress bar Component**: Reviewed and updated documentation and code style.

- **Progress Bar Component**: Renamed `currentValue` prop to `value`.
- **Standardized Prop Types**: Changed the props types of various components to use the standardized interface.

## Why these changes?

- **Clarity**: Improved documentation ensures developers understand the components better, speeding up the development process.
- **Code Quality**: Aligning to standard coding styles enhances readability and maintainability.
- **Scalability**: Using standardized prop types ensures compatibility and eases future feature integrations.

## How to Test

1. Navigate to each updated component and ensure it behaves as expected.
2. Check that all component documentation is clear and free of errors.
3. Ensure that the renaming of `currentValue` to `value` in the Progress Bar component does not break existing functionality.

## Additional Notes

- Please pay close attention to the renamed `value` field in the Progress Bar component to ensure backward compatibility.
